olp-cpp-sdk-core/include/olp/core/cache/CacheSettings.h

```diff
@@ -113,6 +122,18 @@ struct CORE_API CacheSettings {
    */
   EvictionPolicy eviction_policy = EvictionPolicy::kLeastRecentlyUsed;
 
+  /**
+   * @brief This flag sets the compression policy to be applied on the database.
+   *
+   * Even if the input data is incompressible, the kSnappyCompression
+   * implementation will efficiently detect that and will switch to uncompressed
+   * mode. In some cases though, when the data to be inserted is already
+   * compressed by any means, e.g. protobuf or other serialization protocols, it
+   * might not be worth it to enable any compression at all as it will eat up
+   * some CPU. The default value is CompressionType::kSnappyCompression.
+   */
+  CompressionType compression = CompressionType::kSnappyCompression;
```

Comment:
That makes the cache abstraction leaking. And might fire back with not the best-looking API.

Reply:
Any suggestions what would be better?

Reply:
I advise against this option. 
First off, this "Even if the input data is incompressible, the kSnappyCompression implementation will efficiently detect that and will switch to uncompressed mode. [...]  it might not be worth it to enable any compression at all as it will eat up some CPU." contradicts itself. We either trust in Snappy's ability to detect incompressible data, or we don't and invest time in proper API design.
Second, there are some benchmarks[1] indicating that without compression, LevelDB works even slower.
Third, the original discussion that by disabling snappy we gain some perf was unfortunately void wasn't evidence-based.

--
[1] http://www.lmdb.tech/bench/microbench/benchmark.html

Reply:
This benchmark http://www.lmdb.tech/bench/microbench/benchmark.html uses compressible data to measure if it makes sense or not, but in most of our cases, the data you download from the platform is already compressed, e.g. protobuf. And based on my benchmarking we definitely gain a lot of performance increase by disabling compression as the compression threads are still working in the background consuming CPU.

Reply:
As discussed, changed to something more generic.